### PR TITLE
Defer will-change to make the first paint faster.

### LINKED ISSUE #43: Defer applying `will-change` to speed up initial paint
`will-change: transform` seems to be adding an additional 30-50ms of paint to `<gaia-fast-list>` first render. If we can defer that till a few frames later, we may be able to show the user some content faster.

### PR DESCRIPTION
**Pull request by etiennesegonzac**
Fixes #43

### CODE REVIEW

**Comment by wilsonpage** [comment]
In music-nga songs view without the patch I'm seeing a single 65ms paint. With the patch I'm seeing 50ms paint + 40ms recalc style + 50ms paint. Although we may see the painted content a few ms faster, we'll be blocking whatever mainthread task is scheduled to run next, so I'm now not sure if we should land this.

**Comment by etiennesegonzac** [comment]
Updated with @Cwiiis suggestion.
I know remember we did the same for the homescreen :)

File: fast-list.js [hunk 1]
```diff
@@ -209,6 +210,10 @@ FastList.prototype = {
       for (var j = endIndex; j >= startIndex; --j) renderItem(j);
     }
 
+    if (prepareForScroll) {
+      setTimeout(this._prepareForScroll.bind(this), 360);
```
Comment (wilsonpage) [inline fast-list.js:214]: Is there a particular reason for the value `360`?

**Comment by wilsonpage** [comment]
NOTE: I believe the height of the `<ul>` may also impact paint times. But this will be avoided with the initial `overflow: hidden`.

File: fast-list.js (hunk 1 above)
Comment (etiennesegonzac) [inline fast-list.js:214]: It's the same as the "direct protection window" from the DomScheduler.

**Comment by etiennesegonzac** [comment]
Closing this one as this landed as part of the caching/offset work.

**Pull request closed without merging**